pkg/reconciler/pipelinerun/pipelinerun_updatestatus_test.go

```diff
@@ -668,3 +681,789 @@ func TestUpdatePipelineRunStatusFromRuns(t *testing.T) {
+					},
+					OwnerReferences: []metav1.OwnerReference{{UID: prUID}},
+				},
+			}},
+			expectedStatusCRs: childRefsPRStatusData.simpleRun,
+			expectedStatusRuns: map[string]*v1beta1.PipelineRunRunStatus{
+				"pr-run-6-xxyyy": {
+					PipelineTaskName: "run-6",
+					Status:           &v1alpha1.RunStatus{},
+				},
+			},
+		}, {
+			prName:   "orphaned-taskruns-pr",
+			prStatus: func() v1beta1.PipelineRunStatus { return prStatusWithOrphans },
+			trs:      allTaskRuns,
+			runs: []*v1alpha1.Run{{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pr-run-6-xxyyy",
+					Labels: map[string]string{
+						pipeline.PipelineTaskLabelKey: "task-6",
+					},
+					OwnerReferences: []metav1.OwnerReference{{UID: prUID}},
+				},
+			}},
+			expectedStatusTRs: taskRunsPRStatusData.recovered,
+			expectedStatusCRs: childRefsPRStatusData.recovered,
+			expectedStatusRuns: map[string]*v1beta1.PipelineRunRunStatus{
+				"pr-run-6-xxyyy": {
+					PipelineTaskName: "task-6",
+					Status:           &v1alpha1.RunStatus{},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		for _, embeddedVal := range valuesForEmbeddedStatus {
+			t.Run(fmt.Sprintf("%s with %s embedded status", tc.prName, embeddedVal), func(t *testing.T) {
+				ctx := context.Background()
+				cfg := config.NewStore(logtesting.TestLogger(t))
+				cfg.OnConfigChanged(&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName()},
+					Data: map[string]string{
+						embeddedStatusFeatureFlag: embeddedVal,
+					},
+				})
+				ctx = cfg.ToContext(ctx)
+				logger := logtesting.TestLogger(t)
+
+				pr := &v1beta1.PipelineRun{
+					ObjectMeta: metav1.ObjectMeta{Name: tc.prName, UID: prUID},
+					Status:     tc.prStatus(),
+				}
+
+				updatePipelineRunStatusFromSlices(ctx, logger, pr, tc.trs, tc.runs)
+
+				actualPrStatus := pr.Status
+
+				// The TaskRun keys for recovered taskruns will contain a new random key, appended to the
+				// base name that we expect. Replace the random part so we can diff the whole structure
+				actualTaskRuns := actualPrStatus.PipelineRunStatusFields.TaskRuns
+				if actualTaskRuns != nil {
+					fixedTaskRuns := make(map[string]*v1beta1.PipelineRunTaskRunStatus)
+					re := regexp.MustCompile(`^[a-z\-]*?-task-[0-9]`)
+					for k, v := range actualTaskRuns {
+						newK := re.FindString(k)
+						fixedTaskRuns[newK+"-xxyyy"] = v
+					}
+					actualPrStatus.PipelineRunStatusFields.TaskRuns = fixedTaskRuns
+				}
+
+				actualChildRefs := actualPrStatus.ChildReferences
+				if len(actualChildRefs) != 0 {
+					var fixedChildRefs []v1beta1.ChildStatusReference
+					re := regexp.MustCompile(`^[a-z\-]*?-(task|run)-[0-9]`)
+					for _, cr := range actualChildRefs {
+						cr.Name = fmt.Sprintf("%s-xxyyy", re.FindString(cr.Name))
+						fixedChildRefs = append(fixedChildRefs, cr)
+					}
+					actualPrStatus.ChildReferences = fixedChildRefs
+				}
+
+				// Sort the ChildReferences to deal with annoying ordering issues.
```

Comment:
rather than sorting the output, I think you can provide extra inputs/flags to cmp.Diff to get it to ignore ordering of some fields.

Reply:
Oh yeaaaah - I just cargo-culted that over from the original `...FromTaskRuns` test blindly.

Reply:
And actually, I'm not sure if `cmpopts.SortSlices` would make much of a difference here - it'd just be moving the same code around to be a parameter rather than standing alone...